ASoC: intel: sof_sdw: add RT711 SDCA card for MTL platform

yongzhi1 opened the pull request:
Enable on-board rt711 based sound card for MTL RVP.

Signed-off-by: Yong Zhi <yong.zhi@intel.com>
Signed-off-by: Uday M Bhat <uday.m.bhat@intel.com>

plbossart commented on sound/soc/intel/boards/sof_sdw.c:
```diff
@@ -325,6 +325,14 @@ static const struct dmi_system_id sof_sdw_quirk_table[] = {
+		.matches = {
+			DMI_MATCH(DMI_PRODUCT_FAMILY, "Intel_mtlrvp"),
+		},
+		.driver_data = (void *)(RT711_JD1 | SOF_SDW_PCH_DMIC),
```
Can you check if the PCH_DMIC is required @yongzhi1? This is an old way of adding dmic information, we now use the dmic_num that's based on the presence of NHLT. Thanks!

yongzhi1 replied:
Ack

plbossart commented on sound/soc/intel/boards/sof_sdw.c:
```diff
@@ -325,6 +325,14 @@ static const struct dmi_system_id sof_sdw_quirk_table[] = {
+		.matches = {
+			DMI_MATCH(DMI_PRODUCT_FAMILY, "Intel_mtlrvp"),
+		},
+		.driver_data = (void *)(RT711_JD1 | SOF_SDW_PCH_DMIC),
```
Also add SOF_SDW_TGL_HDMI, even if we don't have HDMI working now it will be usable when the topology adds HDMI.

yongzhi1 replied:
Ack

yongzhi1 commented:
V2 update:
1. removed SOF_SDW_PCH_DMIC (use options snd_sof_intel_hda_common dmic_num=4 sof_use_tplg_nhlt=1)
2. Added SOF_SDW_TGL_HDMI in quirks.

Tested with tplg @bardliao shared:
localhost ~ # aplay -l
**** List of PLAYBACK Hardware Devices ****
card 0: sofsoundwire [sof-soundwire], device 0: Jack out (*) []
  Subdevices: 1/1
  Subdevice #0: subdevice #0
card 0: sofsoundwire [sof-soundwire], device 4: HDMI1 (*) []
  Subdevices: 1/1
  Subdevice #0: subdevice #0
card 0: sofsoundwire [sof-soundwire], device 5: HDMI2 (*) []
  Subdevices: 1/1
  Subdevice #0: subdevice #0
card 0: sofsoundwire [sof-soundwire], device 6: HDMI3 (*) []
  Subdevices: 1/1
  Subdevice #0: subdevice #0
card 0: sofsoundwire [sof-soundwire], device 7: HDMI4 (*) []
  Subdevices: 1/1
  Subdevice #0: subdevice #0

bardliao commented on sound/soc/intel/boards/sof_sdw.c:
```diff
@@ -325,6 +325,14 @@ static const struct dmi_system_id sof_sdw_quirk_table[] = {
+		.matches = {
+			DMI_MATCH(DMI_PRODUCT_FAMILY, "Intel_mtlrvp"),
+		},
+		.driver_data = (void *)(RT711_JD1 | SOF_SDW_TGL_HDMI),
```
@yongzhi1 Please also check if jack detection work with RT711_JD1. It could be RT711_JD1 or RT711_JD2

yongzhi1 replied:
Yes, JD1 should be the correct one, verified on two setup.
